### PR TITLE
Tax id data

### PR DESCRIPTION
**Pull request by MikeDeberg**

### CODE REVIEW

File: yax/modules/tax_id_data_builder.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    Parameters
+    ----------
+
+
+    Returns
+    -------
+
+
+    """
+    tax_id_data = output
```
Comment (ebolyen) [inline yax/modules/tax_id_data_builder.py]: Currently this needs to be `tax_id_data, = output`

File: yax/modules/tax_id_data_builder.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    tax_id_data.tax_id_data = taxidtool.build_taxid_data(nodes_dmp,
+                                                         names_dmp,
+                                                         gi_taxid_nucl_fp)
+
+    output.complete
```
Comment (ebolyen) [inline yax/modules/tax_id_data_builder.py]: This is a method, and it doesn't need to be called anyways.

File: yax/artifacts/tax_id_data.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from yax.state.type.artifact import Artifact
+from yax.shared.utilities import taxidtool
+
+
+class TaxIDData(Artifact):
+    def __init__(self, completed):
+        self.tax_id_data = None
+
+        if completed:
+            pass
```
Comment (ebolyen) [inline yax/artifacts/tax_id_data.py]: Shouldn't this set `tax_id_data` to something from the `data_dir`?

File: yax/modules/tax_id_data_builder.py (hunk 1 above)
Comment (MikeDeberg) [inline yax/modules/tax_id_data_builder.py]: c/w

File: yax/modules/tax_id_data_builder.py (hunk 2 above)
Comment (MikeDeberg) [inline yax/modules/tax_id_data_builder.py]: removed

File: yax/artifacts/tax_id_data.py (hunk 3 above)
Comment (MikeDeberg) [inline yax/artifacts/tax_id_data.py]: yes, both the exact fp for writing and then for reading here have been updated accordingly

File: yax/artifacts/tax_id_data.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+from yax.shared.utilities import taxidtool
+import os
+
+
+class TaxIDData(Artifact):
+    def __init__(self, completed):
+        self.tax_id_data = None
+
+        if completed:
+            taxidtool.parse_taxid_data(os.path.join(self.data_dir,
```
Comment (ebolyen) [inline yax/artifacts/tax_id_data.py]: surely `taxidtool.parse_taxid_data` returns something?

File: yax/artifacts/tax_id_data.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+                                                    "tax_id_data.yax"))
+
+    def __complete__(self):
+        taxidtool.write_taxid_data(self.tax_id_data,
+                                   os.path.join(self.data_dir,
+                                                "tax_id_data.yax"))
```
Comment (ebolyen) [inline yax/artifacts/tax_id_data.py]: this is duplicated twice, make it a property of the artifact that you can just reference.

File: yax/artifacts/tax_id_data.py (hunk 4 above)
Comment (MikeDeberg) [inline yax/artifacts/tax_id_data.py]: of couse it does, and dont call me Shirley.

File: yax/artifacts/tax_id_data.py (hunk 5 above)
Comment (MikeDeberg) [inline yax/artifacts/tax_id_data.py]: thank you, now it it simply an attribute called tax_id_data_fp

**Pull request merged**